tests/agent/test_grep.py

```
@@ -0,0 +1,94 @@
+    harness = AgentTestHarness(mode="mock")
+    harness.load_mock_responses([
+        '{"action": "complete", "reason": "mock done"}',
+    ])
+    harness.run_agent("随便什么任务", max_iterations=1)
+    # mock 模式不走工具链，只验证不崩溃
+    assert harness.last_result is not None
+
+
+@pytest.mark.integration
+def test_grep_search_with_real_api():
+    """真实 API 测试：grep 搜索函数定义"""
```

Ruff (0.15.10)

[warning] Docstring contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF002)